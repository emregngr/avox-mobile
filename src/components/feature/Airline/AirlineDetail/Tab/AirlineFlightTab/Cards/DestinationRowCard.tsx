import { MaterialCommunityIcons } from '@expo/vector-icons'
import React from 'react'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'

interface DestinationRowCardProps {
  destination: string
  iconColor: string
}

export const DestinationRowCard = ({ destination, iconColor }: DestinationRowCardProps) => (
  <View className="flex-row items-center px-3 py-2 rounded-full overflow-hidden mr-2 mb-2 border border-background-quaternary bg-background-tertiary">
    <MaterialCommunityIcons color={iconColor} name="airplane" size={20} />

    <ThemedText className="ml-1.5" color="text-90" type="body2">
      {destination}
    </ThemedText>
  </View>
)
