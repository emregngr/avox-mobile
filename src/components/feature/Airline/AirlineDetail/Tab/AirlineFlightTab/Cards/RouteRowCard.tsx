import { Ionicons } from '@expo/vector-icons'
import React from 'react'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'

interface RouteRowCardProps {
  destinationIata: string
  iconColor: string
  origin: string
}

export const RouteRowCard = ({ destinationIata, iconColor, origin }: RouteRowCardProps) => (
  <View className="flex-row items-center py-3">
    <View className="flex-row items-center px-3 py-2 rounded-full overflow-hidden bg-background-tertiary">
      <ThemedText className="mr-1.5" color="text-90" type="body1">
        {origin}
      </ThemedText>
      <Ionicons color={iconColor} name="location" size={16} />
    </View>

    <View className="flex-1 mx-2.5 relative">
      <Ionicons
        className="absolute top-[-10] self-center"
        color={iconColor}
        name="airplane"
        size={20}
      />
    </View>

    <View className="flex-row items-center px-3 py-2 rounded-full overflow-hidden bg-background-tertiary">
      <Ionicons color={iconColor} name="location" size={16} />
      <ThemedText className="ml-1.5" color="text-90" type="body1">
        {destinationIata}
      </ThemedText>
    </View>
  </View>
)
