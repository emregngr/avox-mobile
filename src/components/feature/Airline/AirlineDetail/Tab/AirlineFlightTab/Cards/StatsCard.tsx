import { MaterialCommunityIcons } from '@expo/vector-icons'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'

interface StatsCardProps {
  iconColor: string
  iconName: keyof typeof MaterialCommunityIcons.glyphMap
  label: string
  value: number | string
}

export const StatsCard = ({ iconColor, iconName, label, value }: StatsCardProps) => (
  <View className="flex-1 p-4 rounded-xl overflow-hidden items-center bg-background-secondary">
    <MaterialCommunityIcons color={iconColor} name={iconName} size={28} />

    <ThemedText className="my-2" color="text-100" type="h1">
      {value}
    </ThemedText>

    <ThemedText color="text-90" type="body3" center>
      {label}
    </ThemedText>
  </View>
)
