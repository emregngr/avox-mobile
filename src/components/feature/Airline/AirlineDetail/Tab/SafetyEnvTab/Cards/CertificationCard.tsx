import { Ionicons } from '@expo/vector-icons'
import React from 'react'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'

interface CertificationCardProps {
  certification: string
  iconColor: string
}

export const CertificationCard = ({ certification, iconColor }: CertificationCardProps) => (
  <View className="flex-row items-center px-3 py-2 rounded-full overflow-hidden my-1 self-start bg-background-tertiary">
    <Ionicons color={iconColor} name="ribbon" size={16} />

    <ThemedText className="ml-2" color="text-70" type="body2">
      {certification}
    </ThemedText>
  </View>
)
