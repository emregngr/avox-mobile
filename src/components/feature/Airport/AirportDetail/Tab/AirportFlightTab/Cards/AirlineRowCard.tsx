import { Ionicons } from '@expo/vector-icons'
import React from 'react'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'

interface AirlineRowCardProps {
  airline: string
  iconColor: string
}

export const AirlineRowCard = ({ airline, iconColor }: AirlineRowCardProps) => (
  <View className="flex-row items-center py-2">
    <Ionicons color={iconColor} name="airplane" size={20} />

    <ThemedText className="ml-3" color="text-90" type="body2">
      {airline}
    </ThemedText>
  </View>
  )
