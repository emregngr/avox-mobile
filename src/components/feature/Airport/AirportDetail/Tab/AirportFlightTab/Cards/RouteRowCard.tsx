import { Ionicons } from '@expo/vector-icons'
import React from 'react'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'

interface RouteRowCardProps {
  destinationIata: string
  frequency: string | number
  iconColor: string
}

export const RouteRowCard = ({ destinationIata, frequency, iconColor }: RouteRowCardProps) => (
  <View className="flex-row justify-between items-center py-2">
    <View className="flex-row items-center">
      <Ionicons color={iconColor} name="airplane" size={20} />
      <ThemedText className="ml-3" color="text-90" type="body2">
        {destinationIata}
      </ThemedText>
    </View>

    <View className="px-2.5 py-1 rounded-xl overflow-hidden bg-background-tertiary">
      <ThemedText color="text-90" type="body3">
        {frequency}
      </ThemedText>
    </View>
  </View>
)
