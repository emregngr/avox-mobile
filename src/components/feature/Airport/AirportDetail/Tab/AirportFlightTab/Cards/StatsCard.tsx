import { Ionicons } from '@expo/vector-icons'
import React from 'react'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'

interface StatsCardProps {
  iconColor: string
  iconName: keyof typeof Ionicons.glyphMap
  label: string
  value: number | string
}

export const StatsCard = ({ iconColor, iconName, label, value }: StatsCardProps) => (
  <View className="flex-1 p-4 rounded-xl overflow-hidden items-center bg-background-secondary">
    <Ionicons color={iconColor} name={iconName} size={28} />

    <ThemedText className="my-2" color="text-100" type="h1">
      {value}
    </ThemedText>

    <ThemedText
      color="text-90" lineBreakMode="tail" numberOfLines={2}
      type="body3" center
    >
      {label}
    </ThemedText>
  </View>
)
