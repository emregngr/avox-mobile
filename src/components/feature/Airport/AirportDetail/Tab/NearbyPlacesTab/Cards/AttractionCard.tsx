import { Ionicons } from '@expo/vector-icons'
import React from 'react'
import { TouchableOpacity, View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'

interface AttractionCardProps {
  attractionId: number
  attractionName: string
  description: string
  formattedDistance: string
  getDirectionText: string
  handleDirectionPress: () => void
}

export const AttractionCard = ({
  attractionId,
  attractionName,
  description,
  formattedDistance,
  getDirectionText,
  handleDirectionPress,
}: AttractionCardProps) => {
  const { selectedTheme } = useThemeStore()

  const colors = themeColors?.[selectedTheme]

  return (
    <View
      className="p-4 rounded-xl overflow-hidden flex-row justify-between items-center bg-background-secondary"
      key={attractionId}
    >
      <View className="w-[70px] h-[80px] items-center justify-between">
        <View className="w-10 h-10 rounded-full overflow-hidden bg-background-tertiary items-center justify-center">
          <Ionicons color={colors?.onPrimary100} name="location" size={24} />
        </View>

        <View className="px-2.5 py-1 rounded-xl overflow-hidden flex-row items-center bg-background-tertiary">
          <Ionicons color={colors?.onPrimary100} name="car" size={14} />
          <ThemedText className="ml-1" color="text-90" type="body3">
            {formattedDistance}
          </ThemedText>
        </View>
      </View>

      <View className="flex-1 ml-4">
        <ThemedText
          className="mb-2"
          color="text-100"
          ellipsizeMode="tail"
          numberOfLines={1}
          type="body1"
        >
          {attractionName}
        </ThemedText>

        <ThemedText
          className="mb-2"
          color="text-70"
          ellipsizeMode="tail"
          numberOfLines={5}
          type="body2"
        >
          {description}
        </ThemedText>

        <TouchableOpacity
          activeOpacity={0.7}
          className="flex-row items-center self-end px-2.5 py-2 rounded-xl overflow-hidden bg-background-quaternary"
          hitSlop={20}
          onPress={handleDirectionPress}
        >
          <Ionicons color={colors?.onPrimary100} name="location" size={14} />
          <ThemedText className="ml-1" color="text-90" type="body3">
            {getDirectionText}
          </ThemedText>
        </TouchableOpacity>
      </View>
    </View>
  )
}
