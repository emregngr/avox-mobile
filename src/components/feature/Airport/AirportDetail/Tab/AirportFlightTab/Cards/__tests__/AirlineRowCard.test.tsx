import { render } from '@testing-library/react-native'
import React from 'react'

import { AirlineRowCard } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Cards/AirlineRowCard'

jest.mock('@/components/common/ThemedText', () => {
  const { Text } = require('react-native')

  return {
    ThemedText: ({ children, ...props }: { children: string }) => (
      <Text {...props}>{children}</Text>
    ),
  }
})

const mockedDefaultProps = {
  airline: 'Turkish Airlines',
  iconColor: '#E63946',
}

describe('AirlineRowCard Component', () => {
  it('should render the airline name correctly', () => {
    const { getByText } = render(<AirlineRowCard {...mockedDefaultProps} />)

    const airlineText = getByText(mockedDefaultProps.airline)
    expect(airlineText).toBeTruthy()
  })

  it('should pass the correct props to MaterialCommunityIcons', () => {
    const { getByTestId } = render(<AirlineRowCard {...mockedDefaultProps} />)

    const icon = getByTestId('mocked-material-community-icon')

    expect(icon.props.name).toBe('airplane')
    expect(icon.props.size).toBe(20)
    expect(icon.props.color).toBe(mockedDefaultProps.iconColor)
  })

  it('should pass the correct props to ThemedText', () => {
    const { getByText } = render(<AirlineRowCard {...mockedDefaultProps} />)

    const themedText = getByText(mockedDefaultProps.airline)

    expect(themedText.props.color).toBe('text-90')
    expect(themedText.props.type).toBe('body2')
  })
})

describe('AirlineRowCard Component Snapshot', () => {
  it('should render the AirlineRowCard Component successfully', () => {
    const { toJSON } = render(<AirlineRowCard {...mockedDefaultProps} />)

    expect(toJSON()).toMatchSnapshot()
  })
})
