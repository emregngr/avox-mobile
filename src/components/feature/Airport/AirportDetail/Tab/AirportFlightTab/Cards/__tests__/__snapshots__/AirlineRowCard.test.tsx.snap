// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AirlineRowCard Component Snapshot should render the AirlineRowCard Component successfully 1`] = `
<View
  className="flex-row items-center py-2"
>
  <View
    color="#E63946"
    name="airplane"
    size={20}
    testID="mocked-material-community-icon"
  />
  <Text
    className="ml-3"
    color="text-90"
    type="body2"
  >
    Turkish Airlines
  </Text>
</View>
`;
